packages/back-end/src/validators/saved-queries.ts

```diff
@@ -50,13 +50,25 @@ export type dimensionAxisConfiguration = z.infer<
   typeof dimensionAxisConfigurationValidator
 >;
 
-export const dataVizConfigValidator = z.object({
-  title: z.string().optional(),
-  chartType: z.enum(["bar", "line", "area", "scatter"]),
-  xAxis: xAxisConfigurationValidator,
-  yAxis: z.array(yAxisConfigurationValidator).nonempty(),
-  dimension: z.array(dimensionAxisConfigurationValidator).nonempty().optional(),
-});
+export const dataVizConfigValidator = z.discriminatedUnion("chartType", [
+  // Chart types that require xAxis
+  z.object({
+    title: z.string().optional(),
+    chartType: z.enum(["bar", "line", "area", "scatter"]),
+    xAxis: xAxisConfigurationValidator,
+    yAxis: z.array(yAxisConfigurationValidator).nonempty(),
+    dimension: z
+      .array(dimensionAxisConfigurationValidator)
+      .nonempty()
+      .optional(),
+  }),
+  // Big value chart type that doesn't need xAxis
+  z.object({
+    title: z.string().optional(),
+    chartType: z.literal("big-value"),
+    yAxis: z.array(yAxisConfigurationValidator).nonempty(),
+  }),
+]);
```

Comment:
@ahdriel Working on adding a new "Big Number" data viz type (similar to Mode's Big Number) viz.

It doesn't look like EChartsReact has anything similar to this (maybe a `gauge`), but it would require a lot of weird customization.

One big difference here is that there is no `xAxis` as it's not a chart. I've updated the types here to use a discriminatedUnion, but curious if you have opinions/ideas on a different way to handle this.

It's made some checks on the front end a bit painful and I've had to introduce some logic like `requiresXAxis` and `supportsDimensions`.

Curious if you have thoughts/ideas or if you think this is the best approach.